### ISSUE TITLE
Display the away status in the user list sidebar

### ISSUE THREAD

**Issue by MiniDigger** [thelounge/lounge#872]
if someone is set to away, his name should be displayed in gray or something like that to mark that he is not available right now. 
hexchat does it like this:
![http://i.minidigger.me/2017/01/chrome_07_00-51-32.png](http://i.minidigger.me/2017/01/chrome_07_00-51-32.png)

**Comment by MaxLeiter** [comment]
@xpaw had the idea to dim names in the user list (or we could add another user list section like Modes currently have)

**Comment by astorije** [comment]
I like this idea, but I'm not in favor of dimming names because we are starting to consider it for everything: when hovering on a nick, in #815, now this, etc.

Instead, I'd rather stick something next to the nick, be it http://fontawesome.io/icon/clock-o/ or something like @MiniDigger's screenshot, either same color than nick or green/empty similar to Slack:

<img width="39" alt="screen shot 2017-01-07 at 23 42 01" src="https://cloud.githubusercontent.com/assets/113730/21747392/485b1632-d533-11e6-98fc-8235eebd8ef0.png">

**Comment by ElrondMcBong** [comment]
This is reallly a missing feature in my current implementation. I like the idea of @MaxLeiter to move away-users to another user list section.

**Comment by MiniDigger** [comment]
(fixed the original image) 

I like the icon (either clock or just a colored circle) @astorije suggested, maybe we could even change the sorting to display active users over users who are marked as away.

**Comment by dgw** [comment]
The icon would be plenty on its own. No need to make finding a user more difficult by further subdividing the list. (Anecdotally, I've never used an IRC client that used a separate section for away users.)

**Comment by FingerlessGlov3s** [comment]
Any news on this?

**Comment by McInkay** [comment]
> Any news on this?

Nope. If someone ends up doing it, they'll post here to say so. Feel free to pick it up yourself :-)

**Comment by FingerlessGlov3s** [comment]
> Nope. If someone ends up doing it, they'll post here to say so. Feel free to pick it up yourself :-)

I did start looking at the code, need to make sense of it all first :-)

Is all the server code including the calls to the websocket done code in the src folder?
I then presume all the clientside code in client folder?

**Comment by MiniDigger** [comment]
if you plan on working on this and need help with the code, I suggest joining IRC for discussion, way easier to explain stuff there :)

**Comment by FingerlessGlov3s** [comment]
> if you plan on working on this and need help with the code, I suggest joining IRC for discussion, way easier to explain stuff there :)

Cool will do 👍 .
I'll see how it goes before I start asking annoying questions on IRC xD

**Comment by xPaw** [comment]
We have all the relevant away code handling already.

Currently, server part sends a "user list updated" message to clients, and if client has that channel open it requests the updated user list. This is obviously inefficient, and adding away updates to this wouldn't help.

We would need to do differential updates on user list (join/quit/part should ideally just update the list in-place on client). The full list sync would only be needed after client reconnects.

**Comment by Nek-** [comment]
@xPaw can you be specific about what you want - especially icons & visual stuff - so somebody can handle this? :)

**Comment by el-profesor926** [comment]
Maybe a traffic light system beside the nick.

Green (online), orange (auto away), red (not been active in over 7 days) or something like that.

**Comment by termermc** [comment]
I like the idea with the 7 day bit

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by stwalkerster** [comment]
> Maybe a traffic light system beside the nick.
> 
> Green (online), Orange (auto away), Red (not been active in over 7 days) or something like that.

This idea immediately makes me think of the coloured blobs that some other IRC clients have to display modes that some networks have - blue for voice, cyan for hop, green for op, orange for admin, red for owner. 

That aside, I don't think away time or even idle time is exposed by any of the major networks except in a cross-server whois result, and cross-server whois are often _heavily_ throttled. Depending on what the throttle is, it may take minutes, hours, or possibly _days_ to sync up all the away times for a large channel. A simple WHOX query, however, will return all the users here vs away status immediately.

A simple /away or not marker is all that should be needed here, or possibly even just a marker which shows if a user is away, and no marker if they're here? What about something like this?

![Screenshot from 2021-09-02 00-31-37](https://user-images.githubusercontent.com/722710/131758631-fadc3d45-9c48-4680-bdb6-12c6915252ec.png)

(Yes, I know an emoji might not be the best idea, but this is only a 30-second mockup to illustrate an alternative display idea.)

**Comment by el-profesor926** [comment]
oh I like that zZz for away, pretty neat.

**Comment by yaztromo99** [comment]
This is single biggest missing feature for me. Not being able to see the away status quickly in the user list is frustrating when I have to do a whois to find out for one user at a time. Hexchat and Irccloud support this.

**Comment by MaxLeiter** [comment]
please share any feedback you have here: https://github.com/thelounge/thelounge/pull/4550#issuecomment-1112908451

**Comment by alexjn1984** [comment]
Are there any updates on this implementation?
